---
int16[40] hist